src/test/java/seedu/rc4hdb/logic/parser/commandparsers/HideCommandParserTest.java

```diff
@@ -0,0 +1,59 @@
+package seedu.rc4hdb.logic.parser.commandparsers;
+
+import static seedu.rc4hdb.logic.parser.commandparsers.CommandParserTestUtil.assertParseFailure;
+import static seedu.rc4hdb.logic.parser.commandparsers.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.rc4hdb.logic.commands.modelcommands.HideCommand;
+import seedu.rc4hdb.model.resident.fields.ResidentFields;
+
+public class HideCommandParserTest {
+    private final HideCommandParser parser = new HideCommandParser();
+    private final List<String> emptyList = new ArrayList<>();
+    private final List<String> allFields = ResidentFields.FIELDS.stream()
+            .map(String::toLowerCase).collect(Collectors.toList());
+
+    @Test
+    public void parse_emptyArguments_throwsParseException() {
+        assertParseFailure(parser, "", HideCommandParser.INTENDED_USAGE);
+    }
+
+    @Test
+    public void parse_lowerCaseFields_returnsHideCommand() {
+        HideCommand expectedCommand = new HideCommand(List.of("name", "phone"));
+        assertParseSuccess(parser, "name phone", expectedCommand);
+    }
+
+    @Test
+    public void parse_mixedCaseFields_returnsShowCommand() {
+        HideCommand expectedCommand = new HideCommand(List.of("name", "phone"));
+        assertParseSuccess(parser, "nAmE pHoNe", expectedCommand);
+    }
+
+    @Test
+    public void parse_invalidFields_throwsParseException() {
+        assertParseFailure(parser, "height weight", ShowCommandParser.ERROR_MESSAGE);
+    }
+
+    @Test
+    public void parse_validFieldsWithSpecifiers_throwsParseException() {
+        assertParseFailure(parser, "/all name phone", ShowCommandParser.ERROR_MESSAGE);
+    }
```

Comment:
Does this include situations where the user specify both valid and invalid fields within the command?

Reply:
Yes. In this case, `/all` is parsed as a field and the parser throws the correct error message as intended.